plugin.program.autowidget/resources/lib/common/utils.py

```diff
@@ -451,25 +451,27 @@ def write_file(file, content, mode="w"):
 
 def read_json(file, log_file=False, default=None):
     data = None
-    if xbmcvfs.exists(file):
-        with xbmcvfs.File(file) as f:
-            content = six.ensure_text(f.read())
-            try:
-                data = json.loads(content)
-            except (ValueError, TypeError) as e:
-                log("Could not read JSON from {}: {}".format(file, e), level="error")
-                if log_file:
-                    log(content, level="debug")
-                return default
-    else:
+    # path = os.path.join(_addon_data, file) if _addon_data not in file else file
+    path = xbmcvfs.translatePath(file)
+    if not os.path.exists(path):
         log("{} does not exist.".format(file), level="error")
         return default
+    with contextlib.closing(codecs.open(path, "r", encoding="utf-8")) as f:
```

Comment:
This one should be `xbmcvfs.File(...)` as well. Same question about `closing`.

Reply:
if I change it to that the skin doesn't raise an error but never loads the paths. I don't know why

Reply:
I can probably remove closing if you want since that wasn't the cause of the problem. not sure closing does any harm though. the problem seemed to be caused by json.dump direct to a file. Somehow that would result in files being cut off half way